include/robots/control/obstacle_circumnavigation.h

```diff
@@ -187,16 +200,19 @@ class ObstacleCircumnavigator {
         int whichLeaveLine = -1;
 
         const Eigen::Vector2d robotPosition = { robotPose.x().value(), robotPose.y().value() };
-        StraightLine robotLine;
-        if (std::isnan(robotGoal.x().value())) {
-            // No goal specified: assume robot wants to go in a straight line
-            robotLine.m = tan(robotPose.yaw()).value();
-            robotLine.c = robotPose.y().value() - robotLine.m * robotPose.x().value();
-        } else {
-            // Aim for robot's actual goal
-            const Eigen::Vector2d goal = { robotGoal.x().value(), robotGoal.y().value() };
-            robotLine = StraightLine::fromPoints(robotPosition, goal);
-        }
+        const Line2 robotLine = [&]() {
```

Comment:
I think this is a somewhat overkillish use of a lambda - much too easy not too see the () at the end for a start. Either turn it into a one liner (syntactically if not whitespacily) with the ternary or make ``robotLine`` non-const